library/init/lean/elaborator.lean

```diff
@@ -930,6 +930,41 @@ let matchesOnly := match spec.only with
 | some only := n = only.id.val ∨ only.ids.any (λ id, n = id.val) in
 if matchesOnly then some (spec.id.val ++ n) else none
 
+/-- Global name lookup
+
+Accessible global name:
+A name `n` is called an *accessible global name* if the environment contains a
+declaration `n` or `_private.<module>.n` where `<module>` is the name of the
+current module.
+
+Active open/export command:
+A command `open/export ns ...` is *active* if
+* it is declared within the current scope or
+* it is an export command inside a namespace `ns'` and there is an active
+  open/export command for `ns'`.
```

Comment:
As noted in the code, this is Lean 3's behavior, but is a bit weird: it would activate the `export` in
```
namespace A export B end A
open A (a) -- should this not activate `a` and only `a`?
```

Reply:
Yes, we should fix this weird behavior in Lean4.

Reply:
Done. `renaming` and `hiding` should be fine, I suppose.